bb8/src/internals.rs

```diff
@@ -169,7 +169,7 @@ where
                 }
             }
             if let Some(lifetime) = config.max_lifetime {
-                if now - conn.conn.birth >= lifetime {
+                if conn.conn.is_expired(now - lifetime) {
```

Comment:
Nit: would like to see this split out into a separate commit before the main change.

Reply:
Can do!